cg/meta/upload/gt.py

```diff
@@ -43,93 +47,168 @@ def data(self, analysis: Analysis) -> dict:
-    def get_parsed_qc_metrics_data(qc_metrics: Path) -> MIPMetricsDeliverables:
-        """Parse the information from a qc metrics file"""
+    def _get_parsed_qc_metrics_deliverables_mip_dna(
+        qc_metrics_file: Path,
+    ) -> MIPMetricsDeliverables:
+        """Parse and return a QC metrics file."""
         qcmetrics_raw: dict = ReadFile.get_content_from_file(
-            file_format=FileFormat.YAML, file_path=qc_metrics
+            file_format=FileFormat.YAML, file_path=qc_metrics_file
         )
         return MIPMetricsDeliverables(**qcmetrics_raw)
 
-    def upload(self, data: dict, replace: bool = False):
-        """Upload data about genotypes for a family of samples."""
-        self.gt.upload(str(data["bcf"]), data["samples_sex"], force=replace)
+    def _get_samples_sex_raredisease(self, case: Case) -> dict[str, dict[str, str]]:
+        """Return sex information from StatusDB and from analysis prediction."""
+        qc_metrics_file: Path = self._get_qcmetrics_file(case_id=case.internal_id)
+        samples_sex: dict[str, dict[str, str]] = {}
+        for sample in case.samples:
+            sample_id: str = sample.internal_id
+            samples_sex[sample_id] = {
+                "pedigree": sample.sex,
+                "analysis": UploadGenotypesAPI._get_analysis_sex_raredisease(
+                    qc_metrics_file=qc_metrics_file, sample_id=sample_id
+                ),
+            }
+        return samples_sex
 
     @staticmethod
-    def _is_variant_file(genotype_file: File):
-        return genotype_file.full_path.endswith("vcf.gz") or genotype_file.full_path.endswith("bcf")
+    def _get_analysis_sex_raredisease(qc_metrics_file: Path, sample_id: str) -> str:
+        """Return analysis sex for each sample of an analysis."""
+        qc_metrics: list[MetricsBase] = (
+            UploadGenotypesAPI._get_parsed_qc_metrics_deliverables_raredisease(qc_metrics_file)
+        )
+        for metric in qc_metrics:
+            if metric.name == RAREDISEASE_PREDICTED_SEX_METRIC and metric.id == sample_id:
+                return str(metric.value)
 
-    def _get_genotype_files(self, version_id: int) -> list:
-        return self.hk.files(version=version_id, tags=["genotype"]).all()
+    @staticmethod
+    def _get_parsed_qc_metrics_deliverables_raredisease(qc_metrics: Path) -> list[MetricsBase]:
+        """Parse and return a QC metrics file."""
+        qcmetrics_raw: dict = ReadFile.get_content_from_file(
+            file_format=FileFormat.YAML, file_path=qc_metrics
+        )
+        return [MetricsBase(**metric) for metric in qcmetrics_raw["metrics"]]
 
     @staticmethod
-    def is_suitable_for_genotype_upload(case_obj: Case) -> bool:
-        """Check if a cancer case is contains WGS and normal sample."""
+    def _is_variant_file(genotype_file: File):
+        return genotype_file.full_path.endswith(
+            FileExtensions.VCF_GZ
+        ) or genotype_file.full_path.endswith(FileExtensions.BCF)
```

Comment:
Is this method needed anymore? Seems as if the filtering on VCF_GZ and BCF has already been done when this is called

Reply:
It is redundant, removed